cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -331,7 +323,13 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration) ext.Deployment
 			},
 		},
 	}
-	return d
+}
+
+func prependSelfHostedFlock(name string, args []string) []string {
+	// TODO: flock has to be inside the container, and a shell must exist with this approach
+	// Since the flock binary is present at /bin/flock in a busybox image and /usr/bin/flock in a debian image, we can't hardcode the path
```

Comment:
cc @timstclair 
You're trying to reduce the images' size and attack area, right now we're depending on the `flock` binary and a shell for the given reasons